plugin-server/src/cdp/exceptions-manager.ts

```diff
@@ -0,0 +1,114 @@
+import LRUCache from 'lru-cache'
+
+import { Hub, Team } from '../types'
+import { PostgresUse } from '../utils/db/postgres'
+import { HogFunctionInvocationGlobals } from './types'
+
+// Maps a fingerprintfor easy lookup like: { 'team_id:merged_fingerprint': primary_fingerprint }
+type ExceptionFingerprintByTeamType = Record<string, string>
+
+type ExceptionGroup = {
+    mergedFingerprints: string[][]
+    active: boolean
+}
+
+const FINGERPRINT_CACHE_AGE_MS = 60 * 10 * 1000 // 10 minutes
+
+export class ExceptionsManager {
+    fingerprintMappingCache: LRUCache<number, Record<string, ExceptionGroup>> // team_id: { primary_fingerprint: ExceptionGroup }
+
+    constructor(private hub: Hub) {
+        // There is only 5 per team so we can have a very high cache and a very long cooldown
+        this.fingerprintMappingCache = new LRUCache({ max: 1_000_000, maxAge: FINGERPRINT_CACHE_AGE_MS })
+    }
+
+    private async fetchExceptionFingerprintMapping(teams: Team['id'][]): Promise<ExceptionFingerprintByTeamType> {
+        const exceptionFingerprintMapping: ExceptionFingerprintByTeamType = {}
+
+        // Load the cached values so we definitely have them
+        teams.forEach((teamId) => {
+            const cached = this.fingerprintMappingCache.get(teamId)
+
+            if (cached) {
+                Object.entries(cached).forEach(([primaryFingerprint, { mergedFingerprints }]) => {
+                    mergedFingerprints.forEach((mergedFingerprint) => {
+                        exceptionFingerprintMapping[`${teamId}:${mergedFingerprint}`] = primaryFingerprint
+                    })
+                })
+            }
+        })
+
+        const teamsToLoad = teams.filter((teamId) => !this.fingerprintMappingCache.get(teamId))
+
+        if (teamsToLoad.length) {
+            const result = await this.hub.postgres.query(
+                PostgresUse.COMMON_READ,
+                `SELECT fingerprint, merged_fingerprints, team_id, status
+                FROM posthog_errortrackinggroup
+                WHERE team_id = ANY($1) AND merged_fingerprints != '{}'`,
+                [teamsToLoad],
+                'fetchExceptionTrackingGroups'
+            )
```

Comment:
I had a comment here that is not showing because of some line changes: https://github.com/PostHog/posthog/pull/24751#discussion_r1741150152